### PR TITLE
Pdns AXFR, IXFR loglevels

### PR DESCRIPTION
**Pull request by mind04**
### Short description
Decrease the verbosity of AXFR and IXFR in the logs

Inspired by and successor of #9658 This pull will change the log levels for both, primary and secondary.

Closes #9658 #9514

Most XFR related events are now logged at notice level. And will not show up in the logs at the default log level.

This is the only (non error) message still logged at warning level on the primary:
```
Nov 09 13:33:31 AXFR of domain 'example.com' initiated by 127.0.0.2:36543
```

The secondary is only showing this at warning level:
```
Nov 09 13:33:31 AXFR done for 'example.com', zone committed with serial number 2005092501
```

You can argue about these two lines, but in my opinion it is important to not completely hide XFR at default log level.

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by tih** [comment]
I like it! Much more complete than my quick hack in #9658, and I also never thought of using "Notice" actively.

File: pdns/tcpreceiver.cc [hunk 1]
```diff
@@ -544,26 +544,20 @@ int TCPNameserver::doAXFR(const DNSName &target, std::unique_ptr<DNSPacket>& q,
...
   bool narrow;
   bool NSEC3Zone=false;
   if(securedZone && dk.getNSEC3PARAM(target, &ns3pr, &narrow)) {
     NSEC3Zone=true;
     if(narrow) {
-      g_log<<Logger::Error<<"Not doing AXFR of an NSEC3 narrow zone '"<<target<<"' for "<<q->getRemote()<<endl;
-      noAXFRBecauseOfNSEC3Narrow=true;
+      g_log<<Logger::Warning<<"Not doing AXFR of an NSEC3 narrow zone '"<<target<<"' for "<<q->getRemote()<<endl;
+      outpacket->setRcode(RCode::Refused);
+      // FIXME: should actually figure out if we are auth over a zone, and send out 9 if we aren't
```
Comment (Habbie) [inline pdns/tcpreceiver.cc:555]: I thought 9 was reserved for RFC2136

**Review by Habbie — APPROVED** [review]
Approved, with one open question (about a comment that was already there). I've asked the other vendors if they use NOTAUTH like that.

File: pdns/tcpreceiver.cc (hunk 1 above)
Comment (mind04) [reply pdns/tcpreceiver.cc:555]: There is nothing to figure out here. The zone is narrow so it does exist. I will remove the comment.

Comment (Habbie) [reply pdns/tcpreceiver.cc:555]: ah yes, in this context there is nothing to figure out. Thanks :)

**Review by Habbie — APPROVED** [review]

**Pull request merged**